edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -75,7 +77,7 @@ public void InitiateCacheRefresh()
 
         public Task WaitForCacheRefresh(TimeSpan timeout) => this.refreshCacheCompleteSignal.WaitAsync(timeout);
 
-        public async Task RefreshServiceIdentity(string id)
+        public async Task RefreshServiceIdentity(string id, bool invokeServiceIdentitiesUpdated = true)
```

Comment:
Using this bool so that we don't have to update every time if we're calling it from RefreshServiceIdentities, but so that we still will invoke the event if we're not calling from that method. I'm also open to changing this to be a separate method that is called.

Reply:
when would we want this event to not fire?  Seems like having such a mechanism would lead to potential inconsistencies down the line.

Reply:
RefreshServiceIdentities calls this method within a loop of all the identities. And we fire the event after RefreshServiceIdentities completes. 

It seems wasteful to fire the event every time RefreshServiceIdentity is called within RefreshServiceIdentities

Reply:
I would rather prefer not to expose such flag to public if it is only internal details. We can create a private method that both `RefreshServiceIdentity` and `RefreshServiceIdentities` can call.